### DISCUSSION TITLE
`prop classname did not match` with turborepo internal package

### DISCUSSION THREAD

**Discussion by hwookim** [mantinedev/mantine#4301] [Q&A]
I don't speak English very well, so I'm using a translator.
Please forgive me if it's a bit awkward.

The structure of my project is similar to the following example.
[turborepo next starter](https://github.com/vercel/turbo/tree/main/examples/basic)

In my case, I'm using @mantine for the web package and the ui package.
I'm using the ui package by importing it into the web package.
I'm creating a package utilizing @mantine/core Button in ui package and using it in web package.

I have an SSR issue.
Styles are not being loaded on initial loading and then changing later.
With errors like
```
Warning: Prop `className` did not match. Server: "mantine-UnstyledButton-root mantine-Button-root mantine-kbin90" Client: "mantine-UnstyledButton-root mantine-Button-root mantine-p9ck9v"
```

The ui package has the following dependencies.
```json
  "dependencies": {
    "@mantine/core": "^5.10.4",
    "@mantine/hooks": "^5.10.4",
    "@tabler/icons": "^1.119.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
```

Of course, the web package's `_document` handles the ssr.
Components that call mantine directly (without ui package) are getting the styles just fine.

I'm not sure what to look for.
What do you think I should do?

**Comment by rtivital** [comment]
Follow Next.js getting started guide, if that does not help, use mantine-next-teplate as reference

**Reply by hwookim** [reply]
I already have all that code applied.
```ts
// _document.tsx
const getInitialProps = createGetInitialProps();

export default class _Document extends Document {
  static getInitialProps = getInitialProps;
}
```
Is this what you mean?

When I call mantine directly in nextjs project, the styles are being applied with no problem.
But when I call ui, which is an internal package that uses mantine, the problem occurs.

**Reply by hwookim** [reply]
I created an example project to explain my case,
and strangely enough, it works there.

I'm going to make sure that I can see the difference between my example project and the one that's failing, and then we'll talk about it again.
Thanks for your response.
